blockchain/chain_sync/src/sync.rs

```diff
@@ -536,12 +656,25 @@ where
         ))
     }
 
+    /// iterate over headers to make full tipsets and check state accordingly
+    async fn _sync_msgs_check_state(&self, _headers: Vec<BlockHeader>) {
+        todo!()
+    }
```

Comment:
shouldn't this take in a Tipset instead of a vector of headers? Is this going to return a full tipset? Shouldn't this be a slice passed in? Why does it need to be a moved vector?